### DISCUSSION TITLE
PyEnv Maintenance Expectations

### DISCUSSION THREAD

**Discussion by FelicianoTech** [pyenv/pyenv#1771] [General]
Hi,

I wanted to know what are the maintenance expectations and release policy for PyEnv?

PyEnv is a fairly popular project at this point and I'm sure there are many people and organizations who have come to rely on the project. I work at CircleCI and I can say that we proudly use PyEnv to serve Python to our customers.

PRs such as the one for Python v3.8.7 (#1762) has sit for over two weeks now which has a knock on effect for any projects that rely on PyEnv. Timeframe for releases and "the bus factor" comes to mind for me when thinking about this project. The latter is even more important right now while a pandemic is occuring. Is there anything that can be shared regarding this?

Lastly, is there anyway we can help? Either myself personally, CircleCI, or some others in the community?

Respectfully,
Ricardo N Feliciano

**Comment by ushuz** [comment]
Maybe consider adding more maintainers or donating the project to PSF or PyPA? @joshfriend @chrahunt @yyuu

**Comment by gotche** [comment]
I wonder if PRs regarding adding new versions could be auto merged as soon as they passed the tests. I am not referring to about pyenv itself, just adding new python versions.

**Reply by chrahunt** [reply]
The tests being run are for pyenv itself and use a hard-coded set of versions. New versions added do not get tested automatically at this time.

I'm addressing a few PRs that add new versions and stating explicitly the checks I'm doing before approving. That could be spun off into a checklist for reviewers, hopefully making these easier to get out quickly.

**Comment by chrahunt** [comment]
I have reviewed and merged #1762. I don't have answers to your other questions, but no reason that PR should languish.

**Comment by yyuu** [comment]
Sorry for delay 🙇  These days I've been occupied with other stuffs and it's very hard to me to allocate time for maintenance of this project, especially if it's time sensitive. So far this project has been relying (too much) on hard worker contributors like @joshfriend, @blueyed, @chrahunt, et. al, however, it's not always possible to fulfill the maintenance burden in timely manner maybe because popularity of this project.

Donating the project to some body might be an option, however, without having handful of active maintainers, it will just shift a problem of ownership of the project, in my opinion. I am happy to add more maintainers if there is someone who is willing to play the maintainer role.

**Reply by anton-petrov** [reply]
Hi @yyuu ! If you need help, I'm ready to start maintaining the project :)

**Comment by anton-petrov** [comment]
I completely agree with what has been said by @felicianotech
A project of this level (I mean PyEnv) is no longer just a set of scripts, it is an important software component on which a huge number of organizations and people depends.

**Comment by ulgens** [comment]
Is there anything we can do as contributors?
